geth/node/status_node.go

```diff
@@ -215,16 +206,18 @@ func (n *StatusNode) stop() error {
 	n.gethNode = nil
 	n.config = nil
 
-	if err := n.db.Close(); err != nil {
-		return err
+	if n.db != nil {
+		if err := n.db.Close(); err != nil {
+			return err
```

Comment:
It is `nil`ed when doing clean up in `Start()` method and an error occurs.

Reply:
If there is an error in `n.db.Close()` call, `n.db` is not set to `nil`. In `Start()` method, if the node start fails after creating the database and there is an error when closing the database, `n.db` is set to `nil`.

Reply:
Good point!